nuvlaedge/agent/common/status_handler.py

```diff
@@ -40,17 +41,24 @@ def remove_module(self, module_name: str | StatusReport):
         if isinstance(module_name, StatusReport):
             module_name = module_name.origin_module
 
-        self.module_reports.pop(module_name)
+        if module_name in self.module_reports:
+            self.module_reports.pop(module_name)
 
     def process_status(self):
         temp_status = 'UNKNOWN'
         temp_notes = []
 
         def time_diff(date: datetime) -> int:
-            diff: timedelta = datetime.now() - date
-            return diff.seconds
+            diff: float = datetime.now().timestamp() - date.timestamp()
+            return int(diff)
+
+        to_remove = [name for name, report in self.module_reports.items() if time_diff(report.date) > self.STATUS_TIMEOUT]
+        if to_remove:
+            logger.info(f"Removing modules that have not reported in the last {self.STATUS_TIMEOUT} minutes: {', '.join(to_remove)}")
+            _ = [self.module_reports.pop(name) for name in to_remove]
```

Comment:
I think it should work without `_ = ` (not tested)
```suggestion
            [self.module_reports.pop(name) for name in to_remove]
```
did you try ?

Reply:
Yes, that should work. I'll apply the change